src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4599,22 +4598,43 @@ private TypeSymbol MakeNullableOf(TypeWithAnnotations underlying)
 
         public override BoundNode VisitDelegateCreationExpression(BoundDelegateCreationExpression node)
         {
+            Debug.Assert(node.Type.IsDelegateType());
+
             if (node.MethodOpt?.MethodKind == MethodKind.LocalFunction)
             {
                 var syntax = node.Syntax;
                 var localFunc = (LocalFunctionSymbol)node.MethodOpt.OriginalDefinition;
                 ReplayReadsAndWrites(localFunc, syntax, writes: false);
             }
 
-            // The group is skipped by the base call. It should have a default type always
-            if (node.Argument is BoundMethodGroup group)
+            var delegateType = (NamedTypeSymbol)node.Type;
+            switch (node.Argument)
             {
-                VisitRvalue(group.ReceiverOpt);
-                SetAnalyzedNullability(group, default);
-            }
-            else
-            {
-                VisitRvalue(node.Argument);
+                case BoundMethodGroup group:
+                    {
+                        VisitRvalue(group.ReceiverOpt);
+                        SetAnalyzedNullability(group, default);
+                        // https://github.com/dotnet/roslyn/issues/33637: Should update method based on inferred receiver type.
+                        var method = node.MethodOpt;
+                        if (!(method is null) && !group.IsSuppressed)
```

Comment:
>!(method is null) [](start = 28, length = 17)

```method is object```?

Reply:
Seems equivalent.

---
In reply to: [281745786](https://github.com/dotnet/roslyn/pull/35547#discussion_r281745786) [](ancestors = 281745786)